dissect/target/plugins/os/unix/modules.py

```diff
@@ -0,0 +1,71 @@
+from dataclasses import dataclass
+from typing import Iterator, Optional
+
+from dissect.target.helpers.record import TargetRecordDescriptor
+from dissect.target.plugin import Plugin, export, internal
+from dissect.target.target import Target
+
+ModuleRecord = TargetRecordDescriptor(
+    "linux/module",
+    [
+        ("string", "name"),
+        ("varint", "size"),
+        ("varint", "refcount"),
+        ("stringlist", "modules_referred"),
+        ("uri", "source"),
+    ],
+)
+
+
+@dataclass
+class Module:
+    path: Optional[str] = None
+    name: Optional[str] = None
+    size: Optional[int] = None
+    refcnt: Optional[int] = None
+    modules_referred: Optional[list[str]] = None
+
+
+class ModulePlugin(Plugin):
+    def __init__(self, target: Target):
+        super().__init__(target)
+        self._module_paths = list(self.target.fs.glob("/sys/module/*"))
+
+    def check_compatible(self) -> bool:
+        return len(self._module_paths) > 0
+
+    @internal
+    def iterate_modules(self) -> Iterator[Module]:
```

Comment:
```suggestion
    def _iterate_modules(self) -> Iterator[Module]:
```
And rename accordingly where used.